lib/Dancer/Plugin/SimpleCRUD.pm

```diff
@@ -1003,42 +1016,49 @@ SEARCHFORM
         }
     }
 
-    # If we have a query, we need to assemble a WHERE clause...
-    if (length $q) {
-        my ($column_data)
-            = grep { lc $_->{COLUMN_NAME} eq lc params->{searchfield} }
-            @{$columns};
-        debug(
-            "Searching on $column_data->{COLUMN_NAME} which is a "
-            . "$column_data->{TYPE_NAME}"
-        );
-        my $st = params->{searchtype};
+    # If we have a query or a where_filter, we need to assemble a WHERE clause...
+    my $where_filter = $args->{where_filter};
+    if (length $q || $where_filter) {
+        my ($where_filter_sql, @where_filter_binds) = $dbh->generate_where_clauses($where_filter);
+        my @search_wheres;
+        if (length $q) {
+            my ($column_data)
+                = grep { lc $_->{COLUMN_NAME} eq lc params->{searchfield} }
+                @{$columns};
+            debug(
+                "Searching on $column_data->{COLUMN_NAME} which is a "
+                . "$column_data->{TYPE_NAME}"
+            );
+            my $st = params->{searchtype};
 
-        if ($column_data) {
-            my $search_value = $q;
-            if ($st eq 'c' || $st eq 'nc') {
-                $search_value = '%' . $search_value . '%';
-            }
+            if ($column_data) {
+                my $search_value = $q;
+                if ($st eq 'c' || $st eq 'nc') {
+                    $search_value = '%' . $search_value . '%';
+                }
 
-            $query
-                .= " WHERE $table_name."
-                . $dbh->quote_identifier(params->{searchfield})
-                . ($st eq 'c' ? 'LIKE' :
-                   $st eq 'nc' ? 'NOT LIKE' :
-                   $st eq 'ne' ? '!=' : '=')
-                . $dbh->quote($search_value);
-
-            my $matchtype = $st eq "c" ? "contains": 
-                            $st eq "nc" ? "does not contain" :
-                            $st eq "ne" ? "does not equal": "equals";
-            $html
-                .= sprintf(
-                "<p>Showing results from searching for '%s' %s '%s'",
-                encode_entities(params->{searchfield}), $matchtype, encode_entities($q)
-            );
-            $html .= sprintf '&mdash;<a href="%s">Reset search</a></p>',
-                _external_url($args->{dancer_prefix}, $args->{prefix});
+                push(@search_wheres,
+                    "$table_name."
+                    . $dbh->quote_identifier(params->{searchfield})
+                    . ($st eq 'c' ? ' LIKE ' :
+                       $st eq 'nc' ? ' NOT LIKE ' :
+                       $st eq 'ne' ? ' != ' : ' = ')
+                    . $dbh->quote($search_value));
```

Comment:
ISTR that you can't use bind vars when using LIKE / NOT LIKE, but I'm not certain on that right now.